### ISSUE TITLE
PROCESSOR: Species and diameter groups should be scenario-specific

### ISSUE THREAD

**Issue by lbross** [USFS-PNW/Fia-Biosum-Manager#44]
Species groups (fvs_tree, tree_species_groups, tree_species_groups_list) and diameter groups (tree_diam_groups) are currently stored in the master.mdb. These tables should be moved into the scenario_processor_rule_definitions.mdb and be made scenario specific. The tables can initially be populated from the defaults.

We plan to look at this subsequent to the PROCESSOR redesign.

**Comment by lbross** [comment]
From @lpotts 
This will involve an additional dialog to capture the oven dry weight and dry to green ratio.  It will also involve modifying the queries that reference those columns from the tree_species table as well as changing the audit routine that ensures each species + variant has an assigned oven dry weight and dry to green value.
 
I am also wondering if there would be green weight differences , for example, a doug fir on the coastal range and a doug fir in the blue mountains.  Would they have different values or not?

**Comment by lbross** [comment]
From @jsfried 
These numbers are consistent nationwide in the REF_SPECIES table where I believe we are obtaining them. I will be happier when I have complete documentation for the calculation of Dry2Green. ODWgt (lbs/cu ft) is just WOOD_SPGR_GREENVOL_DRYWT * 62.3 [which is the weight of a cubic foot of water]. Come to think of it, based on this equation working out when using just the wood drywt, I am pretty sure that the bark parameters are not used. I still haven’t been able to get the dry2green calculation to work out.

 I see this as a low priority fix for some time in the future when we can get to it (or if there is higher priority stuff to finish that we are high-centered on).

**Comment by lbross** [comment]
From @jsfried:
 I think that dbh and species groups are project wide even though they show up in the processor module. Ideally, these would be stored in processor specific tables.

**Comment by lbross** [comment]
From @jsfried:
I am thinking that a better design to consider would be to put every species in an “other” category by default so that every species can go somewhere (i.e., default tree_species table is fully populated with some value that by convention is reserved for other—heck, it can even be group 1 if it has to be). Having BioSum hang up on these really rare tree instances and requiring the analyst to pay so much attention to every species is counterproductive.

**Comment by lbross** [comment]
Related suggestion from @lpotts as we would be updating the tree groupings interface anyway.

The TREE_SPECIES.FVS_SPECIES 2-character code is used as a key-relationship when performing the tree groupings. Therefore, it is required to have a value.  If the FVS_SPECIES does not have a value then the COMMON_NAME will drop out of the TREE GROUPINGS dialog.   That’s how the process exists today. See section 2.2.1.6 Processor: Tree Species Groups in the FVS VARIANTS AND TREE SPECIES TECHNICAL REFERENCE documentation.                  

Below is the SQL that gets the COMMON_NAMEs for the TREE GROUPINGS dialog:
SELECT DISTINCT s.common_name
FROM tree_spc_groups_temp t,tree_species s, fvs_tree_species fvs
WHERE t.spcd = s.spcd AND
      TRIM(UCASE(t.fvs_variant)) = TRIM(UCASE(s.fvs_variant)) AND
      TRIM(UCASE(fvs.fvs_species)) = TRIM(UCASE(s.fvs_species))
 
This makes sense to do it this way since some TREE_SPECIES.SPCD + TREE_SPECIES.FVS_VARIANT combinations will NOT BE FOUND in the FVS_TREE_SPECIES table. FVS does not recognize some FIA species codes. Therefore, BIOSUM is attempting to ensure each TREE_SPECIES.FVS_SPECIES has a value to then ensure that all COMMON_NAME values will appear in the TREE GROUPINGS dialog.
 
With that being said, maybe this should be revisited and see if there is an easier means of accomplishing the same thing. We could drop the reference to the FVS_TREE_SPECIES altogether and only get distinct COMMON_NAMEs by FVS_VARIANT. Something like what’s listed below:
 
SELECT DISTINCT s.common_name
FROM tree_spc_groups_temp t,tree_species s
WHERE t.spcd = s.spcd AND
      TRIM(UCASE(t.fvs_variant)) = TRIM(UCASE(s.fvs_variant))     
 
In conclusion, part of the reasoning to include the audit was to ensure a more clearer representation table data set in the TREE_SPECIES table.  The FVS_SPECIES was to represent the FVS crosswalk to the FIA SPCD and this audit tried to fill in all the NULL FVS_SPECIES values in the table.

(Moved comment to this issue; More relevant than to issue #58)

**Comment by lbross** [comment]
There are already 2 tables in /db/master.mdb: tree_species_groups and tree_species_groups_list. We need to add the FIA sp_cd to the tree_species_groups_list table. We need to move these tables to the /processor/scenario_processor_rule_definitions.mdb. We need to add a scenario column to both these tables.

The only screen that is affected is the Processor tree species group screen. 

We can also remove the 2nd audit after we implement this.

**Comment by lbross** [comment]
While working on the copy function for tree diameter groups I found that it was not working correctly for move-in costs and the wood values tab. This has been fixed and will be included with this issue.

**Comment by lbross** [comment]
Per @jsfried this issue has been validated in v5.7.9
